corehq/apps/es/management/commands/elastic_sync_multiplexed.py

```diff
@@ -0,0 +1,222 @@
+        check_task_progress(self.es, task_id)
+
+        self.perform_cleanup()
+
+    def get_adapter(self, cname):
+        index_name = get_registry()[cname].index
+        return _DOC_ADAPTERS_BY_INDEX[index_name]
+
+    def get_source_destination_indexes(self, adapter):
+        return adapter.primary.index_name, adapter.secondary.index_name
+
+    def perform_cleanup(self):
+        print("\n\tPerforming Cleanup:")
+
+        print("\nDeleting Tombstones")
+        self.delete_tombstones(self.adapter.secondary)
+
+        print("\nMarking Index eligible to upgrade to primary")
+        self.mark_primary_eligible()
+
+    def mark_primary_eligible(self):
+        # Would be used to mark the secondary index eligible to be primary.
+        # This would be implemented when supporting models are created.
+        pass
+
+    def delete_tombstones(self, secondary_adapter):
+        # This should be replaced by delete_by_query
+        # https://www.elastic.co/guide/en/elasticsearch/reference/5.1/docs-delete-by-query.html
+        # when on ES version >= 5
+        tombstone_ids = self._get_tombstone_ids(secondary_adapter)
+        actions = [BulkActionItem.delete_id(doc_id=t_id)
+        for t_id in tombstone_ids]
+        secondary_adapter.bulk(actions, refresh=True)
+
+    def _get_tombstone_ids(self, secondary_adapter):
+        query = {
+            "query": {
+                "bool": {
+                    "must": {
+                        "match": {
+                            Tombstone.PROPERTY_NAME: True
+                        }
+                    }
+                }
+            },
+            "_source": False,
+        }
+        scroll_iter = secondary_adapter.scroll(query, size=1000)
+        tombstone_ids = [doc['_id'] for doc in scroll_iter]
+        return tombstone_ids
+
+    def start_reindex_in_es(self, source, dest):
+        # https://www.elastic.co/guide/en/elasticsearch/reference/2.4/docs-reindex.html
+        reindex_body = {
+            "source": {
+                "index": source,
+            },
+            "dest": {
+                "index": dest,
+                "op_type": "create",
+                "version_type": "external"
+            },
+            "conflicts": "proceed"
+        }
+        task_id = self.es.reindex(reindex_body, wait_for_completion=False)
+        return task_id
+
+    def cancel_reindex(self, task_id):
+        result = self.es.tasks.cancel(task_id)
+        node_failures = result.get('node_failure', None)
+        if node_failures:
+            raise CommandError(f"No Reindes process with {task_id} found")
+        node_id = task_id.split(':')[0]
+        task_info = result['nodes'][node_id]['tasks'][task_id]
+
+        start_time = datetime.utcfromtimestamp(task_info['start_time_in_millis'] / 1000)
+        running_duration_seconds = task_info['running_time_in_nanos'] / 10**9
+        duration = human_readable_seconds(running_duration_seconds)
```

Comment:
It would be nice to add these attributes to the value returned by the `manager.get_task()` method.

Reply:
After thinking a bit about it.
`_parse_task_result` result does a good job of returning the relevant information, it should be the responsibility of the called to do the transformation not the adapter.
https://github.com/dimagi/commcare-hq/pull/32370/commits/1eee8f955bf63f755ff8771eccac0a1719ea70c1 is how it turned out to be.

Reply:
> responsibility of the called to do the transformation not the adapter

Yeah, looking at your changes I agree. If we end up doing this logic many places later then it can be abstracted at that time.